homeassistant/components/lock/sesame.py

```diff
@@ -25,16 +26,20 @@
 
 
 # pylint: disable=unused-argument
-def setup_platform(hass, config: ConfigType,
-                   add_devices: Callable[[list], None], discovery_info=None):
+@asyncio.coroutine
+def async_setup_platform(
```

Comment:
Don't switch to async_setup_platform. This isn't an async platform. Adding `update_before_add=True` should be the only necessary change.

Reply:
@armills.  Got it, so this would only be needed if the actual update methods (and the underlying update in pysesame) was async, correct?

Reply:
Exactly. You'd need to switch to the async equivalents if you had other async coroutines you needed to call from setup/update on the external lib.